src/lib/shell/commands/WiFi.cpp

```diff
@@ -102,15 +105,172 @@ static CHIP_ERROR WiFiModeHandler(int argc, char ** argv)
     return SetWiFiMode(argv[0]);
 }
 
+static CHIP_ERROR WifiAddNetwork(char * ssid, char * password)
+{
+    if (ConnectivityMgr().GetDriver() == nullptr)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+
+    uint8_t networkIndex;
+    chip::MutableCharSpan debugText;
+    chip::ByteSpan ssidSpan;
+    chip::ByteSpan passwordSpan;
+
+    ssidSpan     = chip::ByteSpan(reinterpret_cast<const uint8_t *>(ssid), strlen(ssid));
+    passwordSpan = chip::ByteSpan(reinterpret_cast<const uint8_t *>(password), strlen(password));
+
+    if (IsSpanUsable(ssidSpan) && IsSpanUsable(passwordSpan))
+    {
+        ChipLogProgress(DeviceLayer, "[Shell] Adding/Updating network %s", ssidSpan.data());
+
+        /* AddOrUpdateNetwork() checks ssid length and password length. The network info is not persistent. */
+        ConnectivityMgr().GetDriver()->AddOrUpdateNetwork(ssidSpan, passwordSpan, debugText, networkIndex);
+
+        return CHIP_NO_ERROR;
+    }
+    else
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+}
+
 static CHIP_ERROR WiFiConnectHandler(int argc, char ** argv)
+{
+    if (ConnectivityMgr().GetDriver() == nullptr)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+
+    /* Command accepts running either without arguments or either with SSID and password */
+    /* Running without arguments implies that "wifi add ssid password" was executed before */
+    if ((argc != 0) && (argc != 2))
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    chip::ByteSpan ssidSpan;
+    bool connectToNetwork = false;
+
+    if (argc == 0)
+    {
+        /* Retrieve previously added SSID */
+        char ssid[DeviceLayer::Internal::kMaxWiFiSSIDLength] = { 0 };
+
+        ConnectivityMgr().GetNetworkSSID((char *) &ssid);
+        ssidSpan = chip::ByteSpan(reinterpret_cast<const uint8_t *>(ssid), strlen(ssid));
```

Comment:
Is `ssid` meant to include a null-terminator?  Then it's too short.

Reply:
Fixed.